Source/WebCore/page/NavigationHistoryEntry.h

```diff
@@ -36,21 +36,25 @@ class JSValue;
 
 namespace WebCore {
 
+class NavigationEntry;
+
 class NavigationHistoryEntry final : public RefCounted<NavigationHistoryEntry>, public EventTarget, public ContextDestructionObserver {
     WTF_MAKE_ISO_ALLOCATED(NavigationHistoryEntry);
 public:
+    static RefPtr<NavigationHistoryEntry> create(ScriptExecutionContext*, const RefPtr<NavigationEntry>, bool isSameDocument, int64_t index);
+
     using RefCounted<NavigationHistoryEntry>::ref;
     using RefCounted<NavigationHistoryEntry>::deref;
 
     const String& url() const { return m_url; };
     const String& key() const { return m_key; };
     const String& id() const { return m_id; };
-    uint64_t index() const { return m_index; };
+    int64_t index() const { return m_index; };
```

Comment:
I think this can maybe better ask the Navigation object for the index, since otherwise it may get get decoupled (this is what chromium does).